$DragonFly: src/gnu/usr.bin/groff/src/roff/nroff/patches/nroff.sh.no_obj.patch,v 1.2 2005/09/09 14:11:04 asmodai Exp $

--- nroff.sh.orig	2005-09-09 15:01:13.000000000 +0200
+++ nroff.sh	2005-09-09 15:04:33.000000000 +0200
@@ -8,19 +8,23 @@
 case "`exec 2>/dev/null ; locale charmap`" in
   UTF-8)
     T=-Tutf8 ;;
-  ISO-8859-1 | ISO-8859-15)
+  ISO*8859-1 | ISO*8859-15)
     T=-Tlatin1 ;;
   IBM-1047)
     T=-Tcp1047 ;;
+  KOI8-R)
+    T=-Tkoi8-r ;;
   *)
     case "${LC_ALL-${LC_CTYPE-${LANG}}}" in
       *.UTF-8)
         T=-Tutf8 ;;
-      iso_8859_1 | *.ISO-8859-1 | *.ISO8859-1 | \
-      iso_8859_15 | *.ISO-8859-15 | *.ISO8859-15)
+      iso_8859_1 | *.ISO*8859-1 | *.ISO8859-1 | \
+      iso_8859_15 | *.ISO*8859-15 | *.ISO8859-15)
         T=-Tlatin1 ;;
       *.IBM-1047)
         T=-Tcp1047 ;;
+      *.KOI8-R)
+        T=-Tkoi8-r ;;
       *)
         case "$LESSCHARSET" in
           utf-8)
@@ -29,6 +33,8 @@
             T=-Tlatin1 ;;
           cp1047)
             T=-Tcp1047 ;;
+          koi8-r)
+            T=-Tkoi8-r ;;
           *)
             T=-Tascii ;;
           esac ;;
@@ -53,7 +59,7 @@
       exit 1 ;;
     -[iptSUC] | -[dMmrno]*)
       opts="$opts $1" ;;
-    -Tascii | -Tlatin1 | -Tutf8 | -Tcp1047)
+    -Tascii | -Tlatin1 | -Tutf8 | -Tcp1047 | -Tkoi8-r)
       T=$1 ;;
     -T*)
       # ignore other devices
